swift/Tests/Blockchains/OntologyTests.swift

```diff
@@ -0,0 +1,75 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import TrustWalletCore
+import XCTest
+
+extension String {
+    public func indexOf<T>(stuff: T) -> Int where T: StringProtocol {
+        let r = range(of: stuff)
+        guard let rr = r else {
+            return -1
+        }
+        return rr.lowerBound.encodedOffset
+    }
+}
+
+class OntologyTests: XCTestCase {
+    
+    func testSignOngBalanceOf() {
+        
+        let input = TWOntologySigningInput.with {
+            $0.contract = "ONG".data(using: .utf8)!
+            $0.method = "balanceOf".data(using: .utf8)!
+            $0.queryAddress = "ANDfjwrUroaVtvBguDtrWKRMyxFwvVwnZD".data(using: .utf8)!
+        }
+        
+        let output = OntologySigner.sign(input: input)
+        let result = output.encoded.hexString
+        
+        XCTAssertEqual(86, result.indexOf(stuff: "1446b1a18af6b7c9f8a4602f9f73eeb3030f0c29b70962616c616e63654f661400000000000000000000000000000000000000020068164f6e746f6c6f67792e4e61746976652e496e766f6b65"))
+    }
+    
+    func testSignOntBalanceOf() {
+        
+        let input = TWOntologySigningInput.with {
+            $0.contract = "ONT".data(using: .utf8)!
+            $0.method = "balanceOf".data(using: .utf8)!
+            $0.queryAddress = "ANDfjwrUroaVtvBguDtrWKRMyxFwvVwnZD".data(using: .utf8)!
+        }
+        
+        let output = OntologySigner.sign(input: input)
+        let result = output.encoded.hexString
+        
+        XCTAssertEqual(0, result.indexOf(stuff: "00d1"))
+        XCTAssertEqual(86, result.indexOf(stuff: "1446b1a18af6b7c9f8a4602f9f73eeb3030f0c29b70962616c616e63654f66140000000000000000000000000000000000000001"))
+    }
+    
+    func testSignOntTransfer() {
+        
+        let input = TWOntologySigningInput.with {
+            $0.contract = "ONT".data(using: .utf8)!
+            $0.method = "transfer".data(using: .utf8)!
+            $0.fromPrivateKey = Data(hexString: "4646464646464646464646464646464646464646464646464646464646464646")!
+            $0.toAddress = OntologyAddress(string: "Af1n2cZHhMZumNqKgw9sfCNoTWu9de4NDn")!.data
+            $0.amount = 1
+            $0.gasPrice = 500
+            $0.gasLimit = 20000
+            $0.payerPrivateKey = Data(hexString: "4646464646464646464646464646464646464646464646464646464646464652")!
```

Comment:
for Trust wallet users, I assume fromPrivateKey and payerPrivateKey is same?

Reply:
in this case, it should be just `privateKey`

Reply:
They may be same. If you want users to pay gas fee for themselves, they are same. Provide both `fromPrivateKey` and  `payerPrivateKey` offers more possibilities.

Reply:
> in this case, it should be just `privateKey`

I know what you mean. This is our test case, we want to ensure it's validity in multi-signature which is a superset of single signature. So, we use two private key  to ensure both of them are works.

specifically, this multi-signature looks like this:

```json
{
   "Version": 0,
   "Nonce": 294002341,
   "GasPrice": 500,
   "GasLimit": 20000,
   "Payer": "APniYDGozkhUh8Tk7pe35aah2HGJ4fJfVd",
   "TxType": 209,
   "Payload": {
      "Code": "00c66b14fbacc8214765d457c8e3f2b5a1d3c4981a2e9d2a6a7cc814feec06b79ed299ea06fcb94abac41aaf3ead76586a7cc8516a7cc86c51c1087472616e736665721400000000000000000000000000000000000000010068164f6e746f6c6f67792e4e61746976652e496e766f6b65"
   },
   "Attributes": [],
   "Sigs": [
      {
         "PubKeys": [
            "031bec1250aa8f78275f99a6663688f31085848d0ed92f1203e447125f927b7486"
         ],
         "M": 1,
         "SigData": [
            "dcfc87a7befcf0504507929f22a33120f41dadd6abc7fcb2b4aefbfc170c51055096dcd83d5f66e556e5e632b06c117644de3fb67152aabba91ccc9cdd1b8813"
         ]
      },
      {
         "PubKeys": [
            "03d9fd62df332403d9114f3fa3da0d5aec9dfa42948c2f50738d52470469a1a1ee"
         ],
         "M": 1,
         "SigData": [
            "c0bea40c4325a5e137ad4074a43462b137caba02dd1ec4192f0dbd1b699db83472fdced7e61dbb4f94c2067c117fa9013e4dc24a93b4a65a12154f9aca6a563a"
         ]
      }
   ],
   "Hash": "19d4d83ec9046c45969d225eeb09a9fdb34eddbf1086d9ee670d2b7980257749",
   "Height": 0
}
```

As for the single signature, it's more simple:

```json
{
   "Version": 0,
   "Nonce": 294002341,
   "GasPrice": 500,
   "GasLimit": 20000,
   "Payer": "APniYDGozkhUh8Tk7pe35aah2HGJ4fJfVd",
   "TxType": 209,
   "Payload": {
      "Code": "00c66b14fbacc8214765d457c8e3f2b5a1d3c4981a2e9d2a6a7cc814feec06b79ed299ea06fcb94abac41aaf3ead76586a7cc8516a7cc86c51c1087472616e736665721400000000000000000000000000000000000000010068164f6e746f6c6f67792e4e61746976652e496e766f6b65"
   },
   "Attributes": [],
   "Sigs": [
      {
         "PubKeys": [
            "031bec1250aa8f78275f99a6663688f31085848d0ed92f1203e447125f927b7486"
         ],
         "M": 1,
         "SigData": [
            "dcfc87a7befcf0504507929f22a33120f41dadd6abc7fcb2b4aefbfc170c51055096dcd83d5f66e556e5e632b06c117644de3fb67152aabba91ccc9cdd1b8813"
         ]
      }
   ],
   "Hash": "19d4d83ec9046c45969d225eeb09a9fdb34eddbf1086d9ee670d2b7980257749",
   "Height": 0
}
```